bsp/src/org/jetbrains/bsp/project/importing/BspProjectResolver.scala

```diff
@@ -72,7 +72,7 @@ class BspProjectResolver extends ExternalSystemProjectResolver[BspExecutionSetti
     reporter.startTask(structureEventId, None, BspBundle.message("bsp.resolver.resolving.build.structure"))
     val targetsEventId = BuildMessages.randomEventId
     reporter.startTask(targetsEventId, Some(structureEventId), BspBundle.message("bsp.resolver.build.targets"))
-    val targetsRequest = server.workspaceBuildTargets()
+    val targetsRequest = server.workspaceReload().thenCompose(_ => server.workspaceBuildTargets())
```

Comment:
You might want to retain backwards compatibility with BSP servers that don't implement this method. We don't have [protocol version negotiation yet](https://github.com/build-server-protocol/build-server-protocol/issues/56), but I guess we could just swallow a "not-implemented" error here.

Reply:
There's a flag for support of this: `BuildServerCapabilities.canReload`